test/int/wizards/variables/stackFrameWizard.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from 'chai';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { THREAD_ID, ExtendedDebugClient } from 'vscode-chrome-debug-core-testsupport';
+import { VariablesScopeName } from './variablesWizard';
+import { ValidatedSet, IValidatedSet } from '../../core-v2/chrome/collections/validatedSet';
+import { singleElementOfArray } from '../../core-v2/chrome/collections/utilities';
+import { logger } from 'vscode-debugadapter';
+interface IVariablesOfScope {
+    scopeName: VariablesScopeName;
+    variables: DebugProtocol.Variable[];
+}
+
+/** Utility functions to operate on the stack straces and stack frames of the debuggee.
+ * It also provides utilities to access the scopes available in a particular stack frame.
+ */
+export class StackFrameWizard {
+    public constructor(private readonly _client: ExtendedDebugClient, private readonly _stackFrame: DebugProtocol.StackFrame) { }
+
+    /** Return a Wizard to interact with the top stack frame of the debuggee of the client */
+    public static async topStackFrame(client: ExtendedDebugClient): Promise<StackFrameWizard> {
+        const stackTraceResponse = await client.send('stackTrace', { threadId: THREAD_ID });
```

Comment:
We already make the "stackTrace" call to the client inside breakpointsAssertions. Why are we making the call again?

Reply:
Oh this is for when we pause on a debugger statement

Reply:
Refactored some of the code to eliminate this duplicated code...